src/transformers/cache_utils.py

```diff
@@ -1339,3 +1339,255 @@ def update_ssm_state(self, layer_idx: int, new_ssm_state: torch.Tensor):
     def reset(self):
         self.conv_states.zero_()
         self.ssm_states.zero_()
+
+
+class OffloadedStaticCache(StaticCache):
+    """
+    Static cache class to be used with `torch.compile(model)` that offloads to the CPU or
+    another device.
+
+    Attributes:
```

Comment:
nit: in our documentation, we place the arguments here (essentially the ones you wrote in the docstring in `__init__`)

Reply:
Moved it over in new commit.